plugin/evm/message/syncable.go

```diff
@@ -14,78 +14,107 @@ import (
 	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
 )
 
-var _ block.StateSummary = &SyncSummary{}
+var _ Syncable = &BlockSyncSummary{}
 
-// SyncSummary provides the information necessary to sync a node starting
+type Syncable interface {
+	block.StateSummary
+	GetBlockNumber() uint64
+	GetBlockHash() common.Hash
+	GetBlockRoot() common.Hash
+}
+
+type SyncableParser interface {
+	ParseFromBytes(summaryBytes []byte, acceptImpl AcceptImplFn) (Syncable, error)
+}
+
+type AcceptImplFn func(Syncable) (block.StateSyncMode, error)
+
+// BlockSyncSummary provides the information necessary to sync a node starting
 // at the given block.
-type SyncSummary struct {
+type BlockSyncSummary struct {
 	BlockNumber uint64      `serialize:"true"`
 	BlockHash   common.Hash `serialize:"true"`
 	BlockRoot   common.Hash `serialize:"true"`
-	AtomicRoot  common.Hash `serialize:"true"`
 
 	summaryID  ids.ID
 	bytes      []byte
-	acceptImpl func(SyncSummary) (block.StateSyncMode, error)
+	acceptImpl AcceptImplFn
+}
+
+type BlockSyncSummaryParser struct{}
+
+func NewBlockSyncSummaryParser() SyncableParser {
+	return &BlockSyncSummaryParser{}
 }
 
-func NewSyncSummaryFromBytes(summaryBytes []byte, acceptImpl func(SyncSummary) (block.StateSyncMode, error)) (SyncSummary, error) {
-	summary := SyncSummary{}
+func (b *BlockSyncSummaryParser) ParseFromBytes(summaryBytes []byte, acceptImpl AcceptImplFn) (Syncable, error) {
+	summary := BlockSyncSummary{}
 	if codecVersion, err := Codec.Unmarshal(summaryBytes, &summary); err != nil {
-		return SyncSummary{}, err
+		return nil, err
```

Comment:
nit a bit out of scope, but whilst we're at changing this line:

```suggestion
		return nil, fmt.Errorf("decoding to block sync summary: %s", err)
```